--- inc/search.function.php	2009-02-25 09:44:26.000000000 -0800
+++ inc/search.function.php	2009-02-25 10:10:43.000000000 -0800
@@ -73,6 +73,11 @@
 			}
 		}
 	}
+	////////// begin modification for custom fields plugin /////////
+	if (isset($_SESSION["glpi_plugin_customfields_installed"]) && $_SESSION["glpi_plugin_customfields_installed"]==1) {
+                $options = plugin_customfields_cleanSearchOption($options, $action);
+        }
+	////////// end modification for custom fields plugin /////////
 
 	return $options;
 }
--- ajax/dropdownMassiveActionField.php	2009-02-25 09:39:41.000000000 -0800
+++ ajax/dropdownMassiveActionField.php	2009-02-25 09:43:24.000000000 -0800
@@ -174,7 +174,16 @@
 							$plugdisplay=$function($_POST["device_type"],$search["table"],$search["field"],$search["linkfield"]);
 						} 
 					} 
-				} 
+				}
+				////////// begin modification for custom fields plugin /////////
+				elseif (substr($search["table"],0,25)=='glpi_plugin_customfields_')
+                                {
+                                        $function='plugin_customfields_MassiveActionsFieldsDisplay';
+                                        if (function_exists($function)){
+                                                $plugdisplay=$function($_POST["device_type"],$search["table"],$search["field"],$search["linkfield"]);
+                                        }
+                                }
+				////////// end modification for custom fields plugin ///////// 
 				if (!$plugdisplay){
 					dropdown($search["table"],$search["linkfield"],1,$_SESSION["glpiactive_entity"]);
 				}
--- plugins/customfields/setup.php	2009-03-10 10:10:43.000000000 -0700
+++ plugins/customfields/setup.php	2009-03-10 10:11:16.000000000 -0700
@@ -41,7 +41,7 @@
 
 // This will be set to true when applying the glpi patch. 
 // See instructions in the patch directory
-define('CUSTOMFIELDS_GLPI_PATCH_APPLIED', false);
+define('CUSTOMFIELDS_GLPI_PATCH_APPLIED', true);
 
 $ACTIVE_CUSTOMFIELDS_TYPES = array();
 $ALL_CUSTOMFIELDS_TYPES = array();
