Index: search.function.php
===================================================================
--- search.function.php	(revision 315)
+++ search.function.php	(working copy)
@@ -642,7 +642,7 @@
 	if (count($SEARCH_ALL)>0)
 		foreach ($SEARCH_OPTION[$type] as $key => $val){
 			// Do not search on Group Name
-			if (is_array($val)){
+			if (is_array($val) && (!isset($val['purpose']) || $val['purpose']=='search')){
 				$FROM.=addLeftJoin($type,$itemtable,$already_link_tables,$SEARCH_OPTION[$type][$key]["table"],$SEARCH_OPTION[$type][$key]["linkfield"]);
 			}
 		}
@@ -819,7 +819,7 @@
 					$first2=true;
 
 					foreach ($SEARCH_OPTION[$type] as $key2 => $val2)
-						if (is_array($val2)){
+						if (is_array($val2) && (!isset($val2['purpose']) || $val2['purpose']=='search')){
 							// Add Where clause if not to be done ine HAVING CLAUSE
                      if (!isset($val2["usehaving"]) && !isset($val2["nosearch"])){
 								$tmplink=$LINK;
